src/main/java/uk/gov/hmcts/reform/prl/controllers/citizen/CaseController.java

```diff
@@ -137,6 +141,33 @@ public CaseData caseUpdate(
         }
     }
 
+    @PostMapping(value = "{caseId}/{eventId}/party-update-ra", consumes = APPLICATION_JSON, produces = APPLICATION_JSON)
+    @Operation(description = "Update party flags for citizen")
+    @ApiResponses(value = {
+        @ApiResponse(responseCode = "200", description = "Updated party flags for citizen"),
+        @ApiResponse(responseCode = "400", description = "Bad Request"),
+        @ApiResponse(responseCode = "401", description = "Unauthorized"),
+        @ApiResponse(responseCode = "404", description = "Not found")})
+    public ResponseEntity<Object> updateCitizenRAflags(
+        @NotNull @RequestBody CitizenPartyFlagsRequest citizenPartyFlagsRequest,
+        @PathVariable("eventId") String eventId,
+        @PathVariable("caseId") String caseId,
+        @RequestHeader(HttpHeaders.AUTHORIZATION) @Parameter(hidden = true) String authorisation,
+        @RequestHeader(PrlAppsConstants.SERVICE_AUTHORIZATION_HEADER) String s2sToken
+    ) {
+        log.info("Inside updateCitizenRAflags controller {}");
```

Comment:
We need to write up functional and integration tests for both of this new API

Reply:
Done